samples/packages/awscdk-typescript/src/package-config.ts

```diff
@@ -278,7 +304,11 @@ export class PackagePlanner implements cdf.Planner<PackageInfraPlanConstructs, P
 
 }
 
-export type PackageInfraConfig = cdf.InfraConfig<PackageGeneralConfig, PackageNetworkConfig, PackageComponentConfig, PackageServiceConfig, PackageRelationConfig>;
+export type PackageComponentConfigChoices = PackageComponentKeySoftwareConfig | PackageComponentMessageQueueConfig | PackageComponentMessageTopicConfig | PackageComponentStorageBucketConfig | PackageComponentStaticWebsiteHostingConfig;
+
+export type PackageServiceConfigChoices = PackageServiceContainerEcsConfig;
+
+export type PackageInfraConfig = cdf.InfraConfig<PackageGeneralConfig, PackageNetworkConfig, PackageComponentConfigChoices, PackageServiceConfigChoices, PackageRelationConfig>;
```

Comment:
This is to use TS's union typing feature to further detail types info so that the schema gen script can take use of the detailed typing information.